File: sentimental-cash/cash.py
from cs50 import get_float
def main():

    while True:
        cents = get_float("Change:")
        if cents > 0:
            break
    cents = int(cents*100)
    #print(f"cents: {cents}")
    quarters = calculate_quarters(cents)
    #print(quarters)
    change = cents % 25
    dimes = calculate_dimes(change)
        #print(dimes)
    change = change % 10
    nickels = calculate_nickels(change)
        #print(nickels)
    pennies = change % 5
        #print(pennies)
    total_coins = quarters + dimes + nickels + pennies
    print(total_coins)

def calculate_quarters(cents):
    quarters = cents//25
    return quarters

def calculate_dimes(cents):
    dimes = cents//10
    return dimes

def calculate_nickels(cents):
    nickels = cents//5
    return nickels

def calculate_pennies(cents):
    pennies = cents//1
    return pennies




if __name__ == "__main__":
    main()

File: cash/cash.c
#include<cs50.h>
#include<stdio.h>
int calculate_quarters(int cents);
int calculate_dimes(int cents);
int calculate_nickels(int cents);
int calculate_pennies(int cents);
int main(void)
{
    int cents;
    do
    {
        cents = get_int("Change Owed:");
    }
    while(cents < 0);
    int quarters = calculate_quarters(cents);
    printf("Quarters:%i\n",quarters);
    int change = cents % 25;
    int dimes = calculate_dimes(change);
    change = change % 10;
    printf("Dimes:%i\n",dimes);
    int nickels = calculate_nickels(change);
    change = change % 5;
    printf("Nickels:%i\n",nickels);
    int pennies = calculate_pennies(change);
    printf("Pennies:%i\n",pennies);
    int total_coins = quarters + dimes + nickels + pennies;
    printf("Coins:%i\n",total_coins);

}
int calculate_quarters(int cents)
{
    int quarters = cents/25;
    return quarters;
}
int calculate_dimes(int cents)
{
    int dimes = cents/10;
    return dimes;
}
int calculate_nickels(int cents)
{
    int nickels = cents/5;
    return nickels;
}
int calculate_pennies(int cents)
{
    int pennies = cents/1;
    return pennies;
}

